### PR TITLE
[extractor/youtube] Extract more metadata for comments

### PR DESCRIPTION
**Pull request by coletdjnz**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Introduces the following fields for comments:
* `author_url` - The url to the comment author's page
* `author_is_verified` - Whether the author is verified on the platform*
* `is_pinned` - Whether the comment is pinned to the top of the comments

\* also adding for general infodict in (in progress, will be another PR).

Closes https://github.com/yt-dlp/yt-dlp/issues/5411
Related: https://github.com/yt-dlp/yt-dlp/issues/7166

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [X] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [X] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🆕🧑‍💻🗨️

This pull request improves the extraction of YouTube comments by adding more information about the comment authors and their comments, such as verification status, uploader status, and pin status. It also refactors the code in `yt_dlp/extractor/youtube.py` to make it more concise and readable. It affects the files `yt_dlp/extractor/common.py` and `yt_dlp/extractor/youtube.py`.

> _We're scraping the comments from YouTube's site_
> _We're adding new fields to make them more bright_
> _`author_url` and `author_is_verified`_
> _Heave away, me hearties, heave away with pride_

### Walkthrough
*  Add new fields to comment dictionary documentation in `InfoExtractor` class ([link](https://github.com/yt-dlp/yt-dlp/pull/7179/files?diff=unified&w=0#diff-b7f6633815316063666fc3ecebb970e68fdcb5ee01beba03d04c01830e55e8a6R317-R321), [link](https://github.com/yt-dlp/yt-dlp/pull/7179/files?diff=unified&w=0#diff-b7f6633815316063666fc3ecebb970e68fdcb5ee01beba03d04c01830e55e8a6L328-R334))
*  Refactor `_extract_comment` function in `YoutubeIE` class to use a dictionary for extracted fields and add new fields from comment renderer ([link](https://github.com/yt-dlp/yt-dlp/pull/7179/files?diff=unified&w=0#diff-b7b9f6790de4427214b61939432e667d95b929d07fd918b9da1a36d7996cc506L3274-R3282), [link](https://github.com/yt-dlp/yt-dlp/pull/7179/files?diff=unified&w=0#diff-b7b9f6790de4427214b61939432e667d95b929d07fd918b9da1a36d7996cc506L3280-R3318))
*  Simplify `extract_thread` function in `_comment_entries` function in `YoutubeIE` class to use the result of `_extract_comment` function and check for `is_pinned` field ([link](https://github.com/yt-dlp/yt-dlp/pull/7179/files?diff=unified&w=0#diff-b7b9f6790de4427214b61939432e667d95b929d07fd918b9da1a36d7996cc506L3352-R3371))

</details>

### CODE REVIEW

File: yt_dlp/extractor/common.py [hunk 1]
```diff
@@ -314,6 +314,11 @@ class InfoExtractor:
                         * "author" - human-readable name of the comment author
                         * "author_id" - user ID of the comment author
                         * "author_thumbnail" - The thumbnail of the comment author
+                        * "author_url" - The url to the comment author's page
+                        * "author_is_verified" - Whether the author is verified
```
Comment (pukkandan) [inline yt_dlp/extractor/common.py:318]: Not sure if author_is_verified is worth a field.

File: yt_dlp/extractor/youtube.py [hunk 2]
```diff
@@ -3271,37 +3271,50 @@ def _extract_comment(self, comment_renderer, parent=None):
...
         # Timestamp is an estimate calculated from the current time and time_text
         time_text = self._get_text(comment_renderer, 'publishedTimeText') or ''
         timestamp = self._parse_time_text(time_text)
 
-        author = self._get_text(comment_renderer, 'authorText')
-        author_id = try_get(comment_renderer,
-                            lambda x: x['authorEndpoint']['browseEndpoint']['browseId'], str)
+        info.update({
+            # FIXME: non-standard, but we need a way of showing that it is an estimate.
+            'time_text': time_text,
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:3290]:
```suggestion
            '_time_text': time_text,
```
so that hopefully nobody starts relying on it.

**Review by pukkandan — APPROVED** [review]
Rest LGTM

File: yt_dlp/extractor/common.py (hunk 1 above)
Comment (coletdjnz) [reply yt_dlp/extractor/common.py:318]: I think its nice to have and is general across other sites too.

Plus I'm adding `channel_is_verified` in a following PR so might as well have it here as well.

**Pull request merged**